sneakers/channels/soundcloudChannel.py

```diff
@@ -3,72 +3,69 @@
+                'song_name': 'Name to be shown for "song" on soundcloud'
+                },
+            'receiving': {
+                'username': 'Username for user to download sound from.',
+                'song_name':'Name of the sound file to be downloaded'
+                }
             }
-        }
 
-maxLength = 44100 * 60 * 180
-# http://help.soundcloud.com/customer/portal/articles/1662265-how-does-my-account-s-upload-limit-work-
-# you can only upload a TOTAL of 180 minutes - after that you'll have to delete some
+    maxLength = 44100 * 60 * 180
+    maxHourly = 120
 
-maxHourly = 120
-# not sure what this actually is
+    def send(self, data):
+        params = self.params['sending']
+        client = soundcloud.Client(
+            client_id=params['ID'],
+            client_secret=params['secret'],
+            username=params['username'],
+            password=params['password']
+        )
 
-################### Functions ###################
+        frames = []
 
-def send(data, params):
-    client = soundcloud.Client(
-        client_id=params['ID'],
-        client_secret=params['secret'],
-        username=params['username'],
-        password=params['password']
-    )
+        for i in data:
+            frames.append(i)
+            frames.append(',')
 
-    frames = []
+        wf = wave.open('output.wav', 'wb')
+        wf.setnchannels(1)
+        wf.setframerate(44100)
+        wf.setsampwidth(2)
+        wf.writeframes(b''.join(frames))
+        wf.close()
 
-    for i in data:
-        frames.append(i)
-        frames.append(',')
+        print("Done creating sound file")
+        track = client.post('/tracks', track={
+            'title': params['song_name'],
+            'sharing':'public',
+            'asset_data': open('output.wav','rb'),
+            'tag_list':'tag1 \"hip hop\"',
+            'downloadable': 'true' })
+        print("Done uploading")
 
-    wf = wave.open('output.wav', 'wb')
-    wf.setnchannels(1)
-    wf.setframerate(44100)
-    wf.setsampwidth(2)
-    wf.writeframes(b''.join(frames))
-    wf.close()
+        os.remove('output.wav')
 
-    print("Done creating sound file")
-    track = client.post('/tracks', track={
-        'title': params['song_name'],
-        'sharing':'public',
-        'asset_data': open('output.wav','rb'),
-        'tag_list':'tag1 \"hip hop\"',
-        'downloadable': 'true' })
-    print("Done uploading")
+        return
 
-    os.remove('output.wav')
-
-    return
-
-def receive(params):
-    urllib.urlretrieve("http://soundcloud.com/" + params['username'] +\
-            "/" + params['song_name']+ "/download", 'file.wav')
-    wf = wave.open('file.wav', 'r')
-    data = wf.readframes(wf.getnframes())
-    return [data]
+    def receive(self):
+        params = self.params['receiving']
+        urllib.urlretrieve("http://soundcloud.com/" + params['username'] +\
+                "/" + params['song_name']+ "/download", 'file.wav')
```

Comment:
I have some concerns around urlencoding here; what if there's a space in the song name? (etc.)